lib/middleware/request_tracker.rb

```diff
@@ -198,13 +266,63 @@ def self.get_data(env, result, timing, request = nil)
     h
   end
 
+  def self.extract_topic_params_deferred_track(env, request)
+    is_topic_view = request.referer&.start_with?("#{Discourse.base_url}/t/")
```

Comment:
oh can we avoid referer here, it feels fragile to me, various ad blocker like hiding this. 

instead... inject one extra header on the client, then there is no need to gsub or anything ... just look for the DEFER VIEW header and DISCOURSE TOPIC ID header